docs/source/introduction.md

```diff
@@ -481,13 +481,17 @@ Not all file formats have dedicated support to store datetime data, like ESRI
 Shapefile. For such formats, or if you require precision > ms, a workaround is to
 convert the datetimes to string.
 
-Timezone information is preserved where possible, however GDAL only represents
-time zones as UTC offsets, whilst pandas uses IANA time zones (via `pytz` or
-`zoneinfo`). This means that dataframes with columns containing multiple offsets
-(e.g. when switching from standard time to summer time) will be written correctly,
-but when read via `pyogrio.read_dataframe()` will be returned as a UTC datetime
-column, as there is no way to reconstruct the original timezone from the individual
-offsets present.
+Be careful when reading columns with mixed timezone offsets using
+`pyogrio.read_dataframe()`. By default such columns will be returned as a UTC datetime
```

Comment:
I would keep the original intro here, i.e. start with explaining how timezone information is supported in general, and only then warn to be careful with mixed offsets.

(can do a push with some proposed rewording)